### PR TITLE
Remove resolveExtension and validateExtension

### PR DESCRIPTION
**Pull request by CodaFi**
Drop these two and simplify the decl validation state machine by removing the signature checking phase.

The first commit was supposed to be included in #27279, but it got lost - probably during the rebase before I pushed.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please test source compatibility

File: lib/Sema/TypeCheckProtocolInference.cpp [hunk 1]
```diff
@@ -188,22 +188,27 @@ AssociatedTypeInference::inferTypeWitnessesViaValueWitnesses(
     // Invalid case.
     if (extendedNominal == nullptr)
       return true;
-
+    
+    // Validate the nominal type being extended.
+    tc.validateDecl(extendedNominal);
```
Comment (slavapestov) [inline lib/Sema/TypeCheckProtocolInference.cpp]: So validating a nominal doesn't do a whole lot anymore. It just calls computeType() which is trivial. Can you humor me and try removing some of these calls?

**Review by slavapestov — APPROVED** [review]

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**